actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -144,17 +144,19 @@ def initialize(parameters = {})
     end
 
     # Returns true if another +Parameters+ object contains the same content and
-    # permitted flag, or other Hash-like object contains the same content. This
-    # override is in place so you can perform a comparison with `Hash`.
-    def ==(other_hash)
-      if other_hash.respond_to?(:permitted?)
-        super
```

Comment:
`ActionController::Parameters` doesn’t explicitly inherit from anywhere, so `super` calls the implementation from `Object`:

```rb
Object.new == Object.new
#=> false
```